pyemma/coordinates/data/_base/iterable.py

```diff
@@ -292,44 +294,80 @@ def __init__(self, it, it_lagged, return_trajindex):
 
     @property
     def n_chunks(self):
         n1 = self._it.n_chunks
         n2 = self._it_lagged.n_chunks
         return min(n1, n2)
 
+    @property
+    def chunksize(self):
+        return min(self._it.chunksize, self._it_lagged.chunksize)
+
     def __len__(self):
         return min(self._it.trajectory_lengths().min(), self._it_lagged.trajectory_lengths().min())
 
     def __iter__(self):
         return self
 
     def __next__(self):
-        return self.next()
+        _skip_too_short_trajs(self._it_lagged, self._sufficently_long_trajectories)
+        self._it._itraj = self._it_lagged._itraj
 
-    def next(self):
         itraj, data = self._it.next()
+        assert itraj in self._sufficently_long_trajectories, itraj
         itraj_lag, data_lagged = self._it_lagged.next()
-
-        while itraj < itraj_lag:
+        assert itraj_lag in self._sufficently_long_trajectories, itraj_lag
+        if itraj < itraj_lag:
+            self._it._select_file(itraj_lag)
             itraj, data = self._it.next()
+            assert not itraj > itraj_lag
         assert itraj == itraj_lag
 
-        if data.shape[0] > data_lagged.shape[0]:
+        if len(data) > len(data_lagged):
             # data chunk is bigger, truncate it to match data_lagged's shape
-            data = data[:data_lagged.shape[0]]
-        elif data.shape[0] < data_lagged.shape[0]:
+            data = data[:len(data_lagged)]
+        elif len(data) < len(data_lagged):
             raise RuntimeError("chunk was smaller than time-lagged chunk (%s < %s), that should not happen!"
-                               % (data.shape[0], data_lagged.shape[0]))
+                               % (len(data), len(data_lagged)))
         if self._return_trajindex:
             return itraj, data, data_lagged
         return data, data_lagged
 
+    next = __next__
+
     def __enter__(self):
         self._it.__enter__()
         self._it_lagged.__enter__()
+        return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self._it.__exit__(exc_type, exc_val, exc_tb)
         self._it_lagged.__exit__(exc_type, exc_val, exc_tb)
+
+
+def _skip_too_short_trajs(it, sufficiently_long_traj_indices):
+    changed = False
+
+    while (it._itraj not in sufficiently_long_traj_indices
+           and it._itraj < it.state.ntraj):
+        changed = True
+        if len(sufficiently_long_traj_indices) == 1:
+            if it._itraj > sufficiently_long_traj_indices[0]:
+                raise StopIteration('no traj long enough.')
+            it._select_file(sufficiently_long_traj_indices[0])
+            break
+
+        idx = (np.abs(np.array(sufficiently_long_traj_indices) - it._itraj)).argmin()
+        if idx + 1 < len(sufficiently_long_traj_indices):
+            next_itraj = sufficiently_long_traj_indices[idx + 1]
+            it._select_file(next_itraj)
+            assert it._itraj == next_itraj
+        else:
+            raise StopIteration('no trajectory long enough.')
```

Comment:
we should think about what is raised when this condition is met in the constructor